src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +14,108 @@ import {
-      (event as HTMLImageElement).src = randomAvatar;
-    },
-    [user.id]
+  const avatarStyles = useMemo(
+    () => ({
+      backgroundImage: pictureUrl ? `url(${pictureUrl})` : undefined,
+      ...avatarStyle,
+    }),
+    [avatarStyle, pictureUrl]
   );
 
-  return useMemo(() => {
-    return (
-      <S.Container style={{ ...containerStyle }}>
-        {/* Hidden image, used to handle error if image is not loaded */}
-        <img
-          src={pictureUrl}
-          onError={(event) => imageErrorHandler(event)}
-          hidden
-          style={{ display: "none" }}
-          alt={user.anonMode ? DEFAULT_PARTY_NAME : user.partyName}
-        />
-        <S.Avatar
-          onClick={() => openUserProfileModal(user)}
-          className={avatarClassName}
-          backgroundImage={pictureUrl}
-          style={{ ...avatarStyle }}
-        />
-
-        {Reactions.map(
-          (reaction, index) =>
-            reactions.find(
-              (r) => r.created_by === user.id && r.reaction === reaction.type
-            ) && (
-              <div key={index} className="reaction-container">
-                <S.Reaction
-                  role="img"
-                  aria-label={reaction.ariaLabel}
-                  className={reaction.name}
-                  reactionPosition={reactionPosition}
-                >
-                  {reaction.text}
-                </S.Reaction>
-
-                {!muteReactions && !isAudioEffectDisabled && (
-                  <audio autoPlay loop>
-                    <source src={reaction.audioPath} />
-                  </audio>
-                )}
-              </div>
-            )
-        )}
-        {messagesToBand && (
-          <div className="reaction-container">
-            <S.ShoutOutMessage
-              role="img"
-              aria-label="messageToTheBand"
-              reactionPosition={reactionPosition}
-            >
-              {messagesToBand.text}
-            </S.ShoutOutMessage>
-          </div>
-        )}
-      </S.Container>
-    );
-  }, [
-    containerStyle,
-    pictureUrl,
-    user,
-    avatarClassName,
-    avatarStyle,
-    messagesToBand,
-    reactionPosition,
-    imageErrorHandler,
-    openUserProfileModal,
-    reactions,
-    muteReactions,
-    isAudioEffectDisabled,
-  ]);
+  const userDisplayName = user.anonMode ? DEFAULT_PARTY_NAME : user.partyName;
+
+  return (
+    <div className="UserProfilePicture" style={containerStyle}>
+      {venueId && (
+        <UserReactions
+          venueId={venueId}
+          user={user}
+          isMuted={isAudioEffectDisabled}
+          reactionPosition={reactionPosition}
+        >
+          {/* @debt can we use src/components/atoms/UserAvatar/UserAvatar.tsx here? Should we? */}
+          <div
+            role="img"
+            aria-label={`${userDisplayName}'s avatar`}
+            className={avatarClasses}
+            style={avatarStyles}
+            onClick={openProfileModal}
+          />
```

Comment:
My opinion is that we should reuse it here. But Maybe we could change the name of `UserProfilePicture`, because it seems to be a little bit more than that. I don't have any suggestions yet though

Reply:
Yeah, I was definitely leaning towards reuse it as well.. but then realised it would need some refactors to support the use case here (which isn't too hard), and then also realised that we kept it as a 'dump component' so that the status would also need to be passed into it (which again, probably isn't a big concern, as we can make that prop optional if it isn't already)

I'll see how much time is left in this sprint after I rework the `useReactions` logic for performance and try and fit this in. If not, we can handle it in a followup PR.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1323/commits/be40ca41c3a1a4d06cc9e5dbc42f854c7d3c76d1 + https://github.com/sparkletown/sparkle/pull/1323/commits/d4946deea7fa550f4c6b2158a2b73f66a76dc06a